### PR TITLE
Day 4 Homework

### PR DESCRIPTION
**Pull request by feladie**

### CODE REVIEW

File: HW04_ch07_ex02.py [hunk 1]
```diff
@@ -21,15 +21,24 @@
...
+	user_input = str(input('Input: '))
+	done = 'done'
+	last_exp = ''
+	while (user_input != done):
+		last_exp = repr(eval(user_input))
+		print(last_exp)	
+		user_input = str(input('Input: '))
+	else: 
+		return last_exp
+	# Tried type(math.pi) as input, but it doesn't work. :(
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: Please talk to me again. I'm still confused.

File: HW04_ch08_ex04.py [hunk 2]
```diff
@@ -34,15 +35,18 @@ def any_lowercase2(s):
...
     """
     for c in s:
         flag = c.islower()
     return flag
 
 
 def any_lowercase4(s):
-    """Explain what is wrong, if anything, here.
+    """Similar to the previous function, this examines each character of the string. 
+        If s ends with 2 upper-case letters but also contains lower-case letters, flag == False,
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: Show me.

File: HW04_string2.py [hunk 3]
```diff
@@ -42,8 +60,26 @@ def not_bad(s):
...
+    b_front = ''
+    b_back = ''
+    if a_length % 2 == 0:
+           a_front = a[:a_length / 2]
+           a_back = a[a_length / 2:]
+    else:
+        a_front = a[:a_length // 2 + 1]
+        a_back = a[a_length // 2 + 1:]
+
+    if b_length % 2 == 0: # Is this allowed? Can I put an if statement after an else statement?
```
Comment (danielsgriffin) [inline HW04_string2.py]: Yup!

File: HW04_ch08_ex05.py [hunk 4]
```diff
@@ -2,3 +2,28 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]:
```
Book Examples:
 OK  'A'     by 3   => got: 'D'     expected: 'D'
  X  'Z'     by 1   => got: '['     expected: 'A'
 OK  'cheer' by 7   => got: 'jolly' expected: 'jolly'
 OK  'melon' by -10 => got: 'cubed' expected: 'cubed'
 OK  'IBM'   by -1  => got: 'HAL'   expected: 'HAL'

Daniel's Tests:
 OK  'abc'   by 1   => got: 'bcd'   expected: 'bcd'
  X  'xyz'   by 1   => got: 'yz{'   expected: 'yza'
 OK  'ABC'   by 1   => got: 'BCD'   expected: 'BCD'
  X  'XYZ'   by 1   => got: 'YZ['   expected: 'YZA'
  X  'Az'    by 1   => got: 'B{'    expected: 'Ba'
  X  'Aa'    by 26  => got: '[{'    expected: 'Aa'
 OK  'abc'   by -1  => got: 'zab'   expected: 'zab'
 OK  'xyz'   by -1  => got: 'wxy'   expected: 'wxy'
  X  'ABC'   by -1  => got: '@AB'   expected: 'ZAB'
 OK  'XYZ'   by -1  => got: 'WXY'   expected: 'WXY'
  X  'Az'    by -1  => got: '@y'    expected: 'Zy'
 OK  'Aa'    by -26 => got: 'Aa'    expected: 'Aa'
```

**Comment by danielsgriffin** [comment]
Good job.

File: HW04_string2.py [hunk 5]
```diff
@@ -30,8 +38,18 @@ def verbing(s):
...
 def not_bad(s):
-    # +++your code here+++
-    return
+    result = ''
+    not_index = s.find('not')
+    bad_index = s.find('bad')
+    if (not_index > 0 and bad_index > 0):
+        if (bad_index > not_index):
+            result += s[:not_index] + 'good' + s[(bad_index + 3):]
+        else:
```
Comment (leftan) [inline HW04_string2.py]: The first else might not be necessary, since you have the second else clause.

**Pull request closed without merging**